### PR TITLE
chore: code styling

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. Sort `Cargo.toml`
2. Remove dead code.
3. Apply code styling suggestions.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1064

### CODE REVIEW

File: src/datanode/Cargo.toml [hunk 1]
```diff
@@ -44,6 +45,7 @@ object-store = { path = "../object-store" }
 pin-project = "1.0"
 prost.workspace = true
 query = { path = "../query" }
+regex = "1.6"
```
Comment (killme2008) [inline src/datanode/Cargo.toml:48]: Do we still need this crate? I found that you remove the `SCHEMA_PATTERN` below.

Comment (WenyXu) [reply src/datanode/Cargo.toml:48]: > Do we still need this crate? I found that you remove the `SCHEMA_PATTERN` below.

Yep, we still need the regex crate.

https://github.com/GreptimeTeam/greptimedb/blob/3a527c0fd59f5e98eda7abef5c54f03bfab0c22f/src/datanode/src/sql/copy_table_from.rs#L146-L151

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by killme2008** [comment]
@waynexia  PTAL

**Review by waynexia — APPROVED** [review]
LGTM

**Comment by waynexia** [comment]
This is blocked by #1136, I'd update here when it's fixed

**Pull request merged**